src_c/imageext.c

```diff
@@ -207,6 +208,21 @@ write_png(const char *file_name, SDL_RWops *rw, png_bytep *rows, int w, int h,
                  PNG_INTERLACE_NONE, PNG_COMPRESSION_TYPE_BASE,
                  PNG_FILTER_TYPE_BASE);
 
+
```

Comment:
```suggestion
`doing = "set pallete";`
```
This should be set here so that when `goto error;` happens, a more informative error is printed